DxDiagram new API (#1112)
DevExpress/devextreme-documentation: api-reference/10 UI Widgets/dxDiagram/3 Methods/reloadContent(itemKey_applyLayout).md

---
---
##### shortDescription
Reloads diagram data from a data source.

##### param(itemKey?): any

> Specifies a data key of an item to reload.

Reviewer: ```suggestion
Specifies the data key of the item to reload.
```

> * If the parameter is specified the widget reloads the corresponding item and every added and deleted items.

Reviewer: Can you clarify what you mean by "every added and deleted items"?

Author: If the parameter is specified, the widget does not search differences in existing items, but it updates the specified item,  inserted items, and deleted items

Reviewer: ```suggestion
* If the parameter is specified, the widget reloads the corresponding item, inserted items and deleted items.
```

> * If the parameter is not specified the widget reloads the entire diagram.

Reviewer: ```suggestion
* If the parameter is not specified, the widget reloads the entire diagram.
```

##### param(applyLayout?): Boolean
Specifies whether the widget applies the [auto layout](/Documentation/ApiReference/UI_Widgets/dxDiagram/Configuration/nodes/autoLayout/) after content is reloaded.
---

> The changes applied by this method are reflected in a diagram history and can be undone and redone by the UI.

Reviewer: ```suggestion
The changes applied by this method are reflected in the diagram history, and can be undone and redone by the UI.
```

> The changes applied by this method are reflected in the diagram history, and can be undone and redone by the UI.

Reviewer: ```suggestion
Changes applied by this method are reflected in the diagram history. You can use the UI to undo and redo these changes.
```
